### ISSUE TITLE
Shouldn't `_src_path` be stored as absolute path?

### ISSUE THREAD

**Issue by ppo** [copier-org/copier#335]
When creating a project using a local path, shouldn't it be stored as absolute path in the `answers_file`?

If created using `copier . /path/to/destination`, it's saved as `_src_path: .`.

One can do it "auto-manually" using `copier "$PWD" /path/to/destination`.

**Comment by Yajo** [comment]
That's not by design currently, but by accident. However it seems to be a good design, because if you share the project and you know that the template resides next to it in `../template`, having a relative path can be helpful.

In any case, if you end up producing a copier-answers file, it means that you support updates. And the requirements for supporting updates is that your source template is a git  repo with tags (otherwise there's no way to generate a smart diff, thus the update itself will be broken), so your use case is probably wrong by design anyway.

There's still the possibility that it's a local git repo. In that case, I'm still not sure what'd be better... 🤔

**Comment by ppo** [comment]
I guess that if you generate it from a local path, the project is not supposed to be portable/used by someone else.

If it must be portable, the template source should be an URL… otherwise it's probably only accessible by you anyway.

It could be useful to have a command to change the template source… with "use at your own risk".

BTW, is it safe to change that, just that, manually in the answers file?

**Comment by Yajo** [comment]
> BTW, is it safe to change that, just that, manually in the answers file?

I think so.

In any case, this is a real issue because even if we want relative paths, those should be relative to the project, not to the CWD.

**Comment by ppo** [comment]
As it's a choice, there should be an option, right?

Like `--template-src="VALUE"`, where the value can be:
- `absolute`
- `relative`
- the value to use

If that option is passed to `update`, Copier uses that instead of `_src_path` and the answers file is updated accordingly.

`absolute|relative`, are based on the `template_src` during creation and on the CWD during updates.

**Comment by lhupfeldt** [comment]
I ran into the same problem.
Storing the path relative to CWD (I got '.')  is not of much use.
Storing the path as both relative to project **and** as absolute would be great, then the project could be moved, either independently from the template, or they could be moved together.
Adding the `--template-src="VALUE"` is also a good idea.
And yes, generally it is a URL (which can be a `file` url), så maybe the `_src_path` should be renamed to `_src_url`?

**Comment by yajo** [comment]
If you want it to be an absolute path, you can pass it as an absolute path, right? Why do we need extra flags for that?

**Comment by sisp** [comment]
I agree with your analysis, @mgab. I think recording a relative template path relative to the project root is strictly better than recording it relative to the current working directory:

1. `copier update` works regardless of the current working directory.
1. Moving the local template or project root independently invalidates the recorded relative template path in both cases, so there's no degradation.

**Comment by lhupfeldt** [comment]
Yes, it has to be relative to the template. This was also mentioned in the beginning of this thread.
Check https://github.com/copier-org/copier/issues/2589. We use a temporary dir and cd'ing around in order get a correct relative path. It is kind of a hack, so it would be great if copier would handle it.
Saving both the relative and absolute path would allow moving subproject relative to the template locally.
I would suggest to first try relative path, then absolute, as absolute would probably only make sense on the computer where the subproject was generated,
